packages/node/src/integrations/undici/index.ts

```diff
@@ -180,18 +182,21 @@ export class Undici implements Integration {
     };
 
     if (shouldAttachTraceData(stringUrl)) {
-      if (span) {
-        const dynamicSamplingContext = getDynamicSamplingContextFromSpan(span);
-        const sentryBaggageHeader = dynamicSamplingContextToSentryBaggageHeader(dynamicSamplingContext);
-
-        setHeadersOnRequest(request, spanToTraceHeader(span), sentryBaggageHeader);
-      } else {
-        const { traceId, sampled, dsc } = scope.getPropagationContext();
-        const sentryTrace = generateSentryTraceHeader(traceId, undefined, sampled);
-        const dynamicSamplingContext = dsc || getDynamicSamplingContextFromClient(traceId, client, scope);
-        const sentryBaggageHeader = dynamicSamplingContextToSentryBaggageHeader(dynamicSamplingContext);
-        setHeadersOnRequest(request, sentryTrace, sentryBaggageHeader);
-      }
+      const { traceId, spanId, sampled, dsc } = {
```

Comment:
l: It feels like maybe we could extract this into a util 🤔 as this appears to be the same as for http? (and maybe others)